### PR TITLE
Enhancement/nwps 1122 publication data model

### PR DESCRIPTION
**Pull request by FranciscoRuizHEE**

### CODE REVIEW

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml [hunk 1]
```diff
@@ -32,22 +32,14 @@ definitions:
...
-            hipposysedit:validators: [optional]
+            hipposysedit:validators: [required]
           /contentBlocks:
             jcr:primaryType: hipposysedit:field
             hipposysedit:multiple: true
             hipposysedit:ordered: true
             hipposysedit:path: hee:contentBlocks
             hipposysedit:type: hippo:compound
-            hipposysedit:validators: [contentblocks-validator]
-          /relatedContent:
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml:43]: I understand that `relatedContent` field isn't mentioned in the spec, but did we agree with Jon on this just to be sure ?

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml [hunk 2]
```diff
@@ -56,37 +48,65 @@ definitions:
...
             hipposysedit:type: hee:pageLastNextReview
+            hipposysedit:validators: [required]
           /rightHandBlocks:
             jcr:primaryType: hipposysedit:field
             hipposysedit:multiple: true
             hipposysedit:ordered: true
             hipposysedit:path: hee:rightHandBlocks
             hipposysedit:type: hippo:compound
             hipposysedit:validators: [contentblocks-validator]
-          /logoGroup:
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml:66]: I understand that the spec doesn't contain `logoGroup` field ? Could we confirm with Jon about its removal just to be sure [as we have this field in most of the other page DocumentTypes] ?

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml [hunk 3]
```diff
@@ -56,37 +48,65 @@ definitions:
...
             hipposysedit:mandatory: false
             hipposysedit:multiple: false
             hipposysedit:ordered: false
-            hipposysedit:path: hee:logoGroup
+            hipposysedit:path: hee:subtitle
             hipposysedit:primary: false
-            hipposysedit:type: hippo:mirror
+            hipposysedit:type: String
             hipposysedit:validators: [optional]
-          /microHero:
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml:75]: I understand that the spec doesn't contain `microHero` field ? Could we confirm with Jon about its removal just to be sure [as we have this field in most of the other page DocumentTypes] ?

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml [hunk 4]
```diff
@@ -56,37 +48,65 @@ definitions:
...
+            hipposysedit:validators: [required]
           /rightHandBlocks:
             jcr:primaryType: hipposysedit:field
             hipposysedit:multiple: true
             hipposysedit:ordered: true
             hipposysedit:path: hee:rightHandBlocks
             hipposysedit:type: hippo:compound
             hipposysedit:validators: [contentblocks-validator]
-          /logoGroup:
+          /string:
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml:59]: I think the node name should be `subtitle`

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml [hunk 5]
```diff
@@ -56,37 +48,65 @@ definitions:
...
+          /publicationType:
+            jcr:primaryType: hipposysedit:field
+            hipposysedit:mandatory: false
+            hipposysedit:multiple: false
+            hipposysedit:ordered: false
+            hipposysedit:path: hee:publicationType
+            hipposysedit:primary: false
+            hipposysedit:type: DynamicDropdown
+            hipposysedit:validators: [required]
+          /publicationtopics:
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml:86]: Could we camelCase this [`publicationTopics`] (including `hipposysedit:path` property) to be inline with other field names ?

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml [hunk 6]
```diff
@@ -56,37 +48,65 @@ definitions:
...
+            jcr:primaryType: hipposysedit:field
+            hipposysedit:mandatory: false
+            hipposysedit:multiple: true
+            hipposysedit:ordered: false
+            hipposysedit:path: hee:publicationtopics
+            hipposysedit:primary: false
+            hipposysedit:type: String
+            hipposysedit:validators: [required]
+          /publicationprofessions:
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-config/namespaces/hee/report.yaml:95]: Again, could we camelCase this [`publicationProfessions`] (including `hipposysedit:path` property) to be inline with other field names ?

File: repository-data/application/src/main/resources/hcm-content/content/documents/administration/valuelists/publicationprofessions.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
+/content/documents/administration/valuelists/publicationprofessions:
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-content/content/documents/administration/valuelists/publicationprofessions.yaml:1]: Could we `snake_case` the ValueList keys [`selection:key`] just to be inline with other ValueLists ?

File: repository-data/application/src/main/resources/hcm-content/content/documents/administration/valuelists/publicationtopics.yaml [hunk 8]
```diff
@@ -0,0 +1,81 @@
+/content/documents/administration/valuelists/publicationtopics:
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-content/content/documents/administration/valuelists/publicationtopics.yaml:1]: Could we `snake_case` the ValueList keys [`selection:key`] just to be inline with other ValueLists ?

File: repository-data/application/src/main/resources/hcm-content/content/documents/administration/valuelists/publicationtype.yaml [hunk 9]
```diff
@@ -0,0 +1,81 @@
+/content/documents/administration/valuelists/publicationtype:
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-content/content/documents/administration/valuelists/publicationtype.yaml:1]: Could we `snake_case` the ValueList keys [`selection:key`] just to be inline with other ValueLists ?

Also, could we pluralise the name [`publicationTypes` for consistency] ?

File: site/components/src/main/resources/META-INF/hst-assembly/overrides/valueListManager.xml [hunk 10]
```diff
@@ -15,48 +15,34 @@
   See the License for the specific language governing permissions and
   limitations under the License.
   -->
```
Comment (smanivasag) [inline site/components/src/main/resources/META-INF/hst-assembly/overrides/valueListManager.xml:17]: Looks like many entries in this file has got removed including the order of entries. Could we put them back and just include entries for `publicationTypes`, `publicationProfessions` and `publicationProfessions ` for clarity?

**Comment by smanivasag** [comment]
Hi @FranciscoRuizHEE  - Thanks for the PR. Here are few points/queries/clarifications:
- Just so we are aware, the spec doesn't seem to indicate `publicationType` field values as global and so we may need to localise it for channels in the future.
- Can't find the `Documents` multi-valued optional field in the DocumentType, is that intentional ? Or did we agree with Jon to use `Downloads card` to accommodate it ?
- Will it be nice to move `Summary` field right below `Page subtitle` to be inline with other ContentTypes ?

Thanks!

**Comment by smanivasag** [comment]
> Hi @FranciscoRuizHEE - Thanks for the PR. Here are few points/queries/clarifications:
> 
> * Just so we are aware, the spec doesn't seem to indicate `publicationType` field values as global and so we may need to localise it for channels in the future.
> * Can't find the `Documents` multi-valued optional field in the DocumentType, is that intentional ? Or did we agree with Jon to use `Download card` to accommodate it ?
> * Will it be nice to move `Summary` field right below `Page subtitle` to be inline with other ContentTypes ?
> 
> Thanks!

Please ignore my comment above about `Documents` field as the ticket indicates that the `Download cards` will be used to accommodate this. Thanks!

**Comment by smanivasag** [comment]
Hi @FranciscoRuizHEE - Thanks for your commit https://github.com/Health-Education-England/hee-cms-platform/pull/271/commits/5a9765f59ad5487c3f479af17f09b3c0156d3882, looks good. Have further made the following commits to address a few things indicated in the commit message:

- feeaa510d325ff9ea9763b0b72ec3cf9f79e7c19
- 5a2810ec3effc155e929e05661e7f071076cc773
- 134134fb50eb15a90365ad64964f13a9ea1adc7c

Could you review ? Thanks

**Comment by FranciscoRuizHEE** [comment]
@smanivasag Now we have confirmed with Kavita, the changes are accurate and they look good.

**Pull request merged**